packages/amplify-provider-awscloudformation/src/permission-boundary/permission-boundary.ts

```diff
@@ -0,0 +1,124 @@
+      context.print.error('The Permission Boundary ARN specified is not a valid IAM Policy ARN');
+    }
+  }
+
+  const isYes = context?.input?.options?.yes;
+  if (required && (isYes || !doPrompt)) {
+    throw new Error('A Permission Boundary ARN must be specified using --permission-boundary');
+  }
+  if (!doPrompt) {
+    // if we got here, the permission boundary is not required and we can't prompt so return undefined
+    return;
+  }
+  const envName = envNameSupplier();
+  const { permissionBoundaryArn } = await prompt<{ permissionBoundaryArn: string }>({
+    type: 'input',
+    name: 'permissionBoundaryArn',
+    message: `Specify an IAM Policy ARN to use as a Permission Boundary for all IAM Roles in the ${envName} environment (leave blank to remove the Permission Boundary configuration):`,
+    default: getPermissionBoundaryArn(envName),
+    validate,
+  });
+  return permissionBoundaryArn;
+};
+
+/**
+ * This function expects to be called during the env add flow BEFORE the local-env-info file is overwritten with the new env
+ * (ie when it still contains info on the previous env)
+ * context.exeInfo.localEnvInfo.envName is expected to have the new env name
+ */
+const rolloverPermissionBoundaryToNewEnvironment = async (context: $TSContext) => {
+  const newEnv = context.exeInfo.localEnvInfo.envName;
+  const headlessPermissionBoundary = await permissionBoundarySupplier(context, { doPrompt: false, envNameSupplier: () => newEnv });
+  // if headless policy specified, apply that and return
+  if (typeof headlessPermissionBoundary === 'string') {
+    setPermissionBoundaryArn(headlessPermissionBoundary, newEnv, context.exeInfo.teamProviderInfo);
+    return;
+  }
+
+  const currBoundary = getPermissionBoundaryArn();
+  // if current env doesn't have a permission boundary, do nothing
+  if (!currBoundary) {
+    return;
+  }
+  // if existing policy is accessible in new env, apply that one
+  if (await isPolicyAccessible(context, currBoundary)) {
+    setPermissionBoundaryArn(currBoundary, newEnv, context.exeInfo.teamProviderInfo);
+    context.print.info(
+      `Permission Boundary ${currBoundary} has automatically been applied to this environment. To modify this, run \`amplify env update\`.`,
+    );
+    return;
+  }
+  // if existing policy policy is not accessible in the new environment, prompt for a new one
+  context.print.warning(`Permission Boundary ${currBoundary} cannot be applied to resources in this environment.`);
+  setPermissionBoundaryArn(
+    await permissionBoundarySupplier(context, { required: true, envNameSupplier: () => newEnv }),
+    newEnv,
+    context.exeInfo.teamProviderInfo,
+  );
+};
+
+const isPolicyAccessible = async (context: $TSContext, policyArn: string) => {
+  const iamClient = await getIAMClient(() => loadConfiguration(context));
+  try {
+    await iamClient.getPolicy({ PolicyArn: policyArn }).promise();
+  } catch (err) {
+    // if there's an error, then the policy wasn't accessible
```

Comment:
My understanding is cross-account policies require assuming a role in the other account and because the CLI doesn't support this we have to consider a policy like that inaccessible (this is exactly what this check is for actually)